File: C programs/34. Find First and Last Position of Element in Sorted Array.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* searchRange(int* nums, int numsSize, int target, int* returnSize){
    *returnSize=2;
    int* ans = (int*)malloc(2*sizeof(int));
    ans[0]=ans[1]=-1;
    if(numsSize==0){

        return ans;
    }

    int left=0, right=numsSize-1,i=0;
    while(left<=right){
        int mid=left+(right-left)/2;
        if(target==nums[mid]){
            ans[0]=mid;
            right=mid-1;

          }
          else if(target>nums[mid]){
              left=mid+1;

          }else if(target<nums[mid]){
              right=mid-1;

          }

    }
   right = numsSize-1;
        while(left<=right){
        int mid=left+(right-left)/2;
        if(target==nums[mid]){
            ans[1]=mid;
            left=mid+1;

          }
          else if(target>nums[mid]){
              left=mid+1;

          }else if(target<nums[mid]){
              right=mid-1;

          }

    }
return ans;

}

File: other languages/34. Find First and Last Position of Element in Sorted Array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        l,r=0,len(nums)-1
        last,first=-1,-1

        while l<=r:
            m=l+(r-l)//2
            if nums[m]==target:
                first=m
                last=m
                while first > 0 and nums[first-1]==target: 
                    first-=1
                while last < len(nums)-1 and nums[last+1]==target:
                    last+=1
                return [first,last]
            elif target > nums[m]:
                l=m+1
            else:
                r=m-1
        return [first,last]                

        
